datalad_service/migrate/versions.py

```diff
@@ -0,0 +1,209 @@
+                            print(
+                                'Error: {} | The key details cannot be determined - examinekey has no length.'.format(fileInfo))
+                    except:
+                        print(
+                            'Error: {} | Unable to execute examinekey on this file!'.format(fileInfo))
+                else:
+                    print('Error: {} | Unable to find the annex uuid.'.format(fileInfo))
+            except:
+                print('Error: {} | Unable to find the filekey associated with this file. It is quite possible that this file is not annexed.'.format(fileInfo))
+
+        print('Success: {}:{} | Successfully generated a list of update objects for this dataset snapshot.'.format(
+            datasetId, tag))
+        # checkout the git-annex branch of datalad repo
+        try:
+            print('Checking out git-annex branch to apply updates...')
+            ds.repo._git_custom_command([], cmd_str='git checkout git-annex')
+            time.sleep(3)
+            print('Current branch: ')
+            ds.run('git branch')
+            # commit the file tree changes
+            for update in updates:
+                # print('update:', update)
+
+                filepath = update['path']
+                # print('filepath:', filepath)
+
+                contents = update['contents']
+                # print('contents:', contents)
+
+                try:
+                    # create / open file with specific filepath
+                    absolutePath = '{}/{}'.format(dsPath, filepath)
+                    # print('absolutePath:', absolutePath)
+
+                    rmet = open(absolutePath, 'w+')
+                    # print('rmet opened successfully.')
+                    try:
+                        # write the contents to the file
+                        rmet.write(contents)
+                        try:
+                            # close the file
+                            rmet.close()
+                            print('Success: {} | Generated s3 version info in corresponding .log.rmet file'.format(
+                                filepath))
+                        except:
+                            print(
+                                'Error: {} | Unable to close the file'.format(filepath))
+                    except:
+                        print('Error: {} | Unable to write the contents of the update to this file. Contents: {}'.format(
+                            filepath, contents))
+                except:
+                    print(
+                        'Error: {} | Unable to create / open this file'.format(filepath))
+        except:
+            print(
+                'Error: {}:{} | Unable to checkout git-annex branch for dataset'.format(datasetId, tag))
+            sys.exit()
+
+        print('Success: {}:{} | Succesfully created all s3 version .log.rmet files for this dataset snapshot!'.format(
+            datasetId, tag))
+        time.sleep(2)
```

Comment:
I'm concerned about the time.sleep requirement - this is not needed with other uses of _git_custom_command in DataLad so I think this may be hiding something else.

Reply:
@nellh true, those are left over from the ds.run() days